### PR TITLE
Convert canvas-extract to TypeScript

### PR DESCRIPTION
**Pull request by Zyie**
##### Description of change

Nice and simple!

### CODE REVIEW

File: packages/canvas/canvas-extract/global.d.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+declare namespace GlobalMixins
+{
+    interface CanvasRenderer
+    {
+        extract: import('@pixi/canvas-extract').CanvasExtract;
```
Comment (bigtimebuddy) [inline packages/canvas/canvas-extract/global.d.ts]: isn't this `plugins.extract`?

Comment (Zyie) [reply packages/canvas/canvas-extract/global.d.ts]: yes, i didnt see that. Looks like i was too eager to use the new Global Mixins!

Comment (Zyie) [reply packages/canvas/canvas-extract/global.d.ts]: fixed

**Comment by bigtimebuddy** [comment]
I actually think your instincts were right about the GlobalMixins, however, I think we need them to define CanvasRender plugins and Renderer plugins.

Currently, this is defined pretty loosely here: https://github.com/pixijs/pixi.js/blob/dev/packages/core/src/AbstractRenderer.ts#L36-L39

That means users don't get typings for doing things like `renderer.plugins.extract.canvas()`. 

We should do this as a follow-up fix for all the plugins.

File: packages/canvas/canvas-extract/src/CanvasExtract.ts [hunk 2]
```diff
@@ -1,6 +1,9 @@
 import { RenderTexture } from '@pixi/core';
 import { CanvasRenderTarget } from '@pixi/utils';
 import { Rectangle } from '@pixi/math';
+import type { CanvasRenderer } from '@pixi/canvas-renderer';
+import type { DisplayObject } from '@pixi/display';
+import type { BaseRenderTexture } from '@pixi/core';
```
Comment (bigtimebuddy) [inline packages/canvas/canvas-extract/src/CanvasExtract.ts]: Can you make sure these dependencies are including in package.json under dependencies?

File: packages/canvas/canvas-renderer/src/CanvasRenderer.ts [hunk 3]
```diff
@@ -13,6 +13,7 @@ import type {
     RenderTexture,
     BaseRenderTexture,
 } from '@pixi/core';
+import type { CanvasExtract } from '@pixi/canvas-extract';
```
Comment (bigtimebuddy) [inline packages/canvas/canvas-renderer/src/CanvasRenderer.ts]: This is not correct. CanvasRenderer.plugins.extract. Also, this creates a circular dependency. CanvasRenderer should know nothing about canvas-extract.

File: packages/canvas/canvas-renderer/src/CanvasRenderer.ts [hunk 4]
```diff
@@ -44,6 +45,7 @@ export class CanvasRenderer extends AbstractRenderer
     public smoothProperty: string;
     public readonly blendModes: string[];
     public renderingToScreen: boolean;
+    public extract: CanvasExtract;
```
Comment (bigtimebuddy) [inline packages/canvas/canvas-renderer/src/CanvasRenderer.ts]: Not needed.

File: packages/canvas/canvas-renderer/src/CanvasRenderer.ts (hunk 3 above)
Comment (eXponenta) [reply packages/canvas/canvas-renderer/src/CanvasRenderer.ts]: It is inessential because `import type` used inside TS only, and striped in build process

Comment (bigtimebuddy) [reply packages/canvas/canvas-renderer/src/CanvasRenderer.ts]: Yes, but the type imports will be used when we start generating types.

**Comment by ivanpopelyshev** [comment]
`plugins` can be moved to GlobalMixins in separate PR , this PR is fine as it is. circular type deps? we have them anyway, we can fix stuff a bit later.

**Comment by bigtimebuddy** [comment]
@Zyie can remove CanvasExtract from the CanvasRenderer? Otherwise this is good to go

**Comment by bigtimebuddy** [comment]
I've deprecated CanvasRenderer#extract as part of this PR. This is an anti-pattern with the rest of the Renderer plugins, which hang off the `plugins` object.

**Pull request merged**